src/mds/ScrubStack.cc

```diff
@@ -117,6 +117,7 @@ int ScrubStack::enqueue(CInode *in, ScrubHeaderRef& header, bool top)
   clog_scrub_summary(in);
 
   kick_off_scrubs();
+  in->mdcache->maybe_eval_stray(in, false);
```

Comment:
This needs to be done as part of scrubbing the inode and not during enqueue.

Reply:
IMO you should iterate and check all the dentries in the stray directories. Just call this for that ?

Reply:
No need to iterate as scrub already could do it, it's just that it was bugged and would go into a race condition which has has been fixed in this PR.